### ISSUE TITLE
Bayesian Committee Machine (BCM)

### ISSUE THREAD

**Issue by byooooo** [kjappelbaum/pyepal#126]
## Feature description
Add option to parallelize training of GPs using BCM method (factorized GPs). Need to consider potential conflict between existing parallelization for multiple objectives.

## Implementation idea

Create _train_BCM()/_predict_BCM() class function for sklearn instance (and GPy instance if applicable)

**Comment by kjappelbaum** [comment]
I think the interesting thing about BCM would be that we could use the online learning implementation based on the Kalman filter

**Comment by kjappelbaum** [comment]
https://github.com/jopago/GPyBCM/blob/master/GPyBCM.py uses GPy, so we could give it a shot

**Comment by byooooo** [comment]
yea we can make this a low priority

**Comment by kjappelbaum** [comment]
wondering what the API would look like, i.e., how we initialize the `PAL` class. The minimum that the user would need to be able to change would be the number of experts and the model BCM/rBCM and somehow we need to know that the user wants to use this. 

so an initial thought would be to add
```
bcm: bool = False
m: int = 6
bcm_model: str = 'rbcm'
```

the downside of this is that the list of arguments would be quite long

**Comment by kjappelbaum** [comment]
@byooooo do you know a benchmark comparing this to low-rank approximation as the SparseGP one can already use with PyePAL via  `GPy`?

**Comment by byooooo** [comment]
yea i think that would make the instantiation cumbersome. maybe instead we can have another function after initialization with PAL that the user can call (in case they want to use BCM) and have the BCM attributes set to None initially in the PAL class.

So it would be something like:
`PALinstance(*args, **kwargs)`
followed by
`PALinstance.BCM(*args, **kwargs)`
where the args could be the ones you listed above

**Comment by byooooo** [comment]
> @byooooo do you know a benchmark comparing this to low-rank approximation as the SparseGP one can already use with PyePAL via `GPy`?

Nope. 😅 maybe should try this ourselves first before implementing this into the package

**Comment by kjappelbaum** [comment]
Yea and GPflow also has a bunch of different approximations, e.g., https://www.jmlr.org/papers/v18/16-579.html
